check-disk/lib/check-disk.go

```diff
@@ -0,0 +1,121 @@
+package checkdisk
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/mackerelio/checkers"
+)
+
+type diskStatus struct {
+	All   uint64 `json:"all"`
+	Used  uint64 `json:"used"`
+	Free  uint64 `json:"free"`
+	Avail uint64 `json:"available"`
+}
+
+var opts struct {
+	Warning      *float64 `short:"W" long:"warning" value-name:"N" description:"Exit with WARNING status if less than N units of disk are free"`
+	Critical     *float64 `short:"C" long:"critical" value-name:"N" description:"Exit with CRITICAL status if less than N units of disk are free"`
+	WarningRate  *float64 `short:"w" long:"warning-rate" value-name:"N" description:"Exit with WARNING status if less than N % of disk are free"`
```

Comment:
I think the behavior of the original command is nice because the user do not confused by which to use -w or -W. Accept as `*string` and parse it.